packages/dds/tree/src/test/shared-tree/sharedTree.bench.ts

```diff
@@ -377,41 +377,73 @@ describe("SharedTree benchmarks", () => {
+		const opsPerCommit = 2;
+		// Number of peers that are generating commits.
+		const peerCounts = [2, 4];
+		// Number of commits that are generated in the amount of time it takes of a single commit to round-trip.
+		// E.g., 10 is equivalent to all of the following (and more):
+		// - generating 5 edits per second with a 2000ms round-trip time
+		// - generating 10 edits per second with a 1000ms round-trip time
+		// - generating 100 edits per second with a 100ms round-trip time
+		const commitCounts = [1, 5, 10];
+		for (const peerCount of peerCounts) {
+			for (const commitCount of commitCounts) {
+				const test = benchmark({
+					type: BenchmarkType.Measurement,
+					title: `for ${commitCount} commits per peer for ${peerCount} peers`,
+					benchmarkFnCustom: async <T>(state: BenchmarkTimer<T>) => {
+						let duration: number;
+						do {
+							// Since this setup one collects data from one iteration, assert that this is what is expected.
+							assert.equal(state.iterationsPerBatch, 1);
+							const provider = new TestTreeProviderLite(peerCount, factory);
 
-						// Setup
-						const provider = new TestTreeProviderLite(nbPeers, factory);
-						for (let iCommit = 0; iCommit < nbCommits; iCommit++) {
-							for (let iPeer = 0; iPeer < nbPeers; iPeer++) {
-								const peer = provider.trees[iPeer];
-								insert(peer.checkout, 0, `p${iPeer}c${iCommit}`);
+							// This is the start of the stream of commits.
+							// Earlier commits are less out of date and therefore not representative.
+							for (let iCommit = 0; iCommit < commitCount; iCommit++) {
+								for (let iPeer = 0; iPeer < peerCount; iPeer++) {
+									const peer = provider.trees[iPeer];
+									insert(peer.checkout, 0, `p${iPeer}c${iCommit}`);
+								}
 							}
-						}
 
-						// Measure
-						const before = state.timer.now();
-						provider.processMessages();
-						const after = state.timer.now();
-						// Divide the duration by the number of peers so we get the average time per peer.
-						duration = state.timer.toSeconds(before, after) / nbPeers;
-					} while (state.recordBatch(duration));
-				},
-				// Force batch size of 1
-				minBatchDurationSeconds: 0,
-			});
+							// This block generates commits that are all out of date to the same degree
+							for (let iCommit = 0; iCommit < commitCount; iCommit++) {
+								for (let iPeer = 0; iPeer < peerCount; iPeer++) {
+									provider.processMessages(opsPerCommit);
+									const peer = provider.trees[iPeer];
+									insert(peer.checkout, 0, `p${iPeer}c${iCommit}`);
+								}
+							}
+
+							// This block measures commits that are all out of date to the same degree.
+							// We could theoretically measure the time it takes for a single commit to be processed,
+							// but averaging over multiple commits gives a more stable result.
+							let timeSum = 0;
+							for (let iCommit = 0; iCommit < commitCount; iCommit++) {
```

Comment:
If the purpose of this loop is to enable more stable averaging, we should probably iterate some fixed number of times instead of using `commitCount`.